### ISSUE TITLE
Order of fields in XML model can break contact creation

### ISSUE THREAD

**Issue by abbyad** [medic/cht-core#1494]
If the `<clinic>` field is after the `<contact>` in the contact creation XForm then it gets the following error:
`GET https://lg.dev.medicmobile.org/medic/PARENT?_nonce=1446151485472 404 (Not Found)`

eg this works

```
        <data id="NFAM" version="2015-10-01">
          <clinic>
            <contact>NEW</contact>
            ...
          </clinic>
          <contact>
            ...
            <parent>PARENT</parent>
          </contact>
```

and this gets the error

```
        <data id="NFAM" version="2015-10-01">
          <contact>
            ...
            <parent>PARENT</parent>
          </contact>
          <clinic>
            <contact>NEW</contact>
            ...
          </clinic>
```

**Comment by abbyad** [comment]
This is specific our custom work using Enketo, but labeling as Enketo and Documentation for future reference/searches.

**Comment by mrjones-plip** [comment]
I don't think this is an issue any more given that we've updated Enketo - gonna risk closing trusting that @jkuester will keep me honest 😅

**Comment by jkuester** [comment]
Reopening this because I tested and confirmed it is still an issue.  I did not see the 404, but something does go wrong with setting up the parent lineage for the contacts and they are not able to replicate.

@mrjones-plip since you are assigned to this, I will be expecting a PR with the fix!  😅

**Comment by jkuester** [comment]
Looking a bit deeper, it does appear that cht-conf handles making sure the xml elements are in the correct order to avoid this issue (regardless of how you have things ordered in the xlsx file.  So, the _impact_ of this issue is very low (really you should always use cht-conf to convert/upload your forms).  

The sad fact is that all our logic around contact forms is very rigid and brittle and makes a ton of assumptions about the structure of the form.  I am not sure this should still be considered a "Help wanted" issue, but if someone does want to have a crack at fixing it, I would start by looking at the logic in the [contact-save service](https://github.com/medic/cht-core/blob/master/webapp/src/ts/services/contact-save.service.ts) and see if we can make the logic around setting the lineage more generic/streamlined....

**Comment by mrjones-plip** [comment]
Awesome - thanks for the second look at this - much appreciated @jkuester
